pkg/kubelet/cm/devicemanager/pod_devices.go

```diff
@@ -321,10 +328,69 @@ func (pdev *podDevices) deviceRunContainerOptions(podUID, contName string) *Devi
 			annotationsMap[k] = v
 			opts.Annotations = append(opts.Annotations, kubecontainer.Annotation{Name: k, Value: v})
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DevicePluginCDIDevices) {
+			// Updates for CDI devices.
+			cdiDevices := getCDIDeviceInfo(resp, allCDIDevices)
+			opts.CDIDevices = append(opts.CDIDevices, cdiDevices...)
+		}
+	}
+
+	// Although the CDI devices are expected to be empty when this feature is disabled, we still
+	// guard this with a feature gate to avoid any potential issues.
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DevicePluginCDIDevices) {
+		cdiAnnotations := getCDIAnnotations(podUID+"-"+contName, sets.List[string](allCDIDevices), annotationsMap)
```

Comment:
I'm still hung up on this resourceID of `"podUID+"-"+contName`. What is the annotation that will be generated from this? Do we as an end user even care what this, as long as the machinery below knows how to interpret this and inject the proper devices? Can we avoid passing this in and just generate a unique uuid within the body of `getCDIAnnotations`?

Reply:
The unit tests show the annotations that are generated:
```
cdi.k8s.io/devicemanager_podUID-contName
```
The container runtimes such as containerd or crio that would interpret these annotations do not parse these fields and only look at the `cdi.k8s.io/` prefix and the values to get the list of devices. The ID is only needed to ensure that there are no collisions at a container level.

If you think a UUID is better, I'm happy to use that instead, but would then ask for guidance on how this can be generated deterministically for testing purposes.

Reply:
For the record I have no strong opinions here and I'll be OK with whatever @klueska (who knows the CDI area in general much better) prefers